### ISSUE TITLE
Initial refactoring for limit orders

### ISSUE THREAD

**Issue by woodenfurniture** [shapeshift/web#7908]
### Overview

**TL;DR - move swapper specific stuff to higher order components**

### References and additional details

Originally spiked in https://github.com/shapeshift/web/issues/7905

The following changes are likely required to prepare the `TradeInput` component and its children for reuse with limit orders:

- Compose the existing `TradeInputHeader`, `TradeInputBody`, `ConfirmSummary` components without `Acknowledgement` wrappers and `CollapsibleQuoteList`, and pass all feature-specific callbacks and data in as props.
- General cleanup - Move functions like `getQuoteErrorTranslation` to a shared utilities file.
- Within `TradeInputHeader`, `TradeInputBody`, `ConfirmSummary` components, replace `useAppSelector` and `useAppDispatch` calls with props where state or logic is not shared.
- Likely have to make the "cog" menu button contents (on the top-right) a prop too, since limit orders will have different items.

### Acceptance Criteria

Task list above is completed.
